dex/lexi/datum.go

```diff
@@ -0,0 +1,86 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package lexi
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/decred/dcrd/wire"
+)
+
+// datum is a value in the key-value database, along with information about
+// its index entries.
+type datum struct {
+	version byte
+	indexes [][]byte
+	v       []byte
+}
+
+func (d *datum) bytes() ([]byte, error) {
+	if d.version != 0 {
+		return nil, fmt.Errorf("unknown datum version %d", d.version)
+	}
+	bLen := 1 + len(d.v) + wire.VarIntSerializeSize(uint64(len(d.v))) + wire.VarIntSerializeSize(uint64(len(d.indexes)))
+	for _, ib := range d.indexes {
+		bLen += len(ib) + wire.VarIntSerializeSize(uint64(len(ib)))
+	}
+	b := bytes.NewBuffer(make([]byte, 0, bLen))
```

Comment:
Oops. I forgot to doc this encoding stuff. It's pretty much like `BuildyBytes`, but I'm pre-allocating the buffer and allowing any blob size. And I'm leveraging `wire` for the var int stuff.

Reply:
When reading the doc I noticed this:

https://github.com/decred/dcrdex/blob/917902468ae940a359161dac3826b3bcbb633c62/dex/encode/passbytes.go#L24

"...satisfies the json.Unmarshaler interface..."

I think it is the json.Marshaller interface. I could be wrong.

Reply:
Tests all PASS on my machine.